JsonFile.js

```diff
@@ -15,46 +19,100 @@ const DEFAULT_OPTIONS = {
   space: 2,
 };
 
+// A promisified writeFileAtomic
+const wfap = (file, data) =>
+  new Promise((resolve, reject) => {
+    writeFileAtomic(file, data, err => {
+      if (err) reject();
+      else resolve();
+    });
+  });
+
+const lockWrapper = fn => async (file, ...args) => {
```

Comment:
I have considered that but by creating a lockWrapper there is no way for me to extract the file name for the lock file. Perhaps something like
```
callWithLock(() =>
  updateAsync(this.file, key, value, this._getOptions(options)), 'fileName'
);
```